front/src/modules/ui/view-bar/components/FilterDropdownMultipleEntitySearchSelect.tsx

```diff
@@ -0,0 +1,113 @@
+import { useState } from 'react';
+
+import {
+  EntitiesForMultipleEntitySelect,
+  MultipleEntitySelectBase,
+} from '@/ui/input/relation-picker/components/MultipleEntitySelectBase';
+import { EntityForSelect } from '@/ui/input/relation-picker/types/EntityForSelect';
+import { useRecoilScopedState } from '@/ui/utilities/recoil-scope/hooks/useRecoilScopedState';
+import { useRemoveFilter } from '@/ui/view-bar/hooks/useRemoveFilter';
+import { useUpsertFilter } from '@/ui/view-bar/hooks/useUpsertFilter';
+import { filterDefinitionUsedInDropdownScopedState } from '@/ui/view-bar/states/filterDefinitionUsedInDropdownScopedState';
+import { selectedOperandInDropdownScopedState } from '@/ui/view-bar/states/selectedOperandInDropdownScopedState';
+
+import { useViewBarContext } from '../hooks/useViewBarContext';
+
+export const FilterDropdownMultipleEntitySearchSelect = <
+  CustomEntityForSelect extends EntityForSelect,
+>({
+  entitiesForSelect,
+}: {
+  entitiesForSelect: EntitiesForMultipleEntitySelect<CustomEntityForSelect>;
+}) => {
+  const { ViewBarRecoilScopeContext } = useViewBarContext();
+
+  const [selectedEntities, setSelectedEntities] = useState<
+    (CustomEntityForSelect & { isChecked: boolean })[]
+  >(() => [
+    ...entitiesForSelect.selectedEntities.map((entity) => ({
+      ...entity,
+      isChecked: true,
+    })),
+    ...entitiesForSelect.entitiesToSelect.map((entity) => ({
+      ...entity,
+      isChecked: false,
+    })),
+  ]);
+
+  const [selectedOperandInDropdown] = useRecoilScopedState(
+    selectedOperandInDropdownScopedState,
+    ViewBarRecoilScopeContext,
+  );
+
+  const [filterDefinitionUsedInDropdown] = useRecoilScopedState(
+    filterDefinitionUsedInDropdownScopedState,
+    ViewBarRecoilScopeContext,
+  );
+
+  const upsertFilter = useUpsertFilter();
+  const removeFilter = useRemoveFilter();
+
+  const handleChange = (enitityId: string, newCheckedValue: boolean) => {
+    setSelectedEntities((prev) =>
```

Comment:
We avoid generic names even in lambdas, better to have `previousSelectedEntities`